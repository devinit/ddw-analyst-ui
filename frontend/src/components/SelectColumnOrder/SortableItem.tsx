import { useSortable } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';
import React, { FunctionComponent } from 'react';
import StyledItem from './StyledItem';

interface SelectColumnProps {
  id: string;
  count?: number;
}

const SortableItem: FunctionComponent<SelectColumnProps> = ({ count, id }) => {
  const { attributes, listeners, setNodeRef, transform, transition } = useSortable({ id });

  const style = {
    transform: CSS.Transform.toString(transform),
    transition: transition?.toString(),
  };

  return (
    <StyledItem
      style={style}
      ref={setNodeRef}
      {...attributes}
      {...listeners}
      data-testid="qb-select-column-order"
    >
      {count ? `${count}. ${id}` : id}
    </StyledItem>
  );
};

export { SortableItem };
