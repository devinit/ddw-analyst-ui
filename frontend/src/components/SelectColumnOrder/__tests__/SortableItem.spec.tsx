import 'jest-styled-components';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SortableItem } from '../SortableItem';

describe('SortableItem', () => {
  test('renders correctly with the required props', () => {
    const renderer = TestRenderer.create(<SortableItem id="test" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders a count when provided', () => {
    const renderer = TestRenderer.create(<SortableItem id="test" count={3} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
