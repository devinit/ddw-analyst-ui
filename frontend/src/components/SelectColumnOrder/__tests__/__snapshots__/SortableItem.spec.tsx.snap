// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableItem renders a count when provided 1`] = `
<div
  aria-describedby=""
  aria-roledescription="sortable"
  className=""
  data-testid="qb-select-column-order"
  role="button"
  style={
    Object {
      "transform": undefined,
      "transition": undefined,
    }
  }
  tabIndex={0}
>
  3. test
</div>
`;

exports[`SortableItem renders correctly with the required props 1`] = `
<div
  aria-describedby=""
  aria-roledescription="sortable"
  className=""
  data-testid="qb-select-column-order"
  role="button"
  style={
    Object {
      "transform": undefined,
      "transition": undefined,
    }
  }
  tabIndex={0}
>
  test
</div>
`;
